### DISCUSSION TITLE
Scaling for markers for the rare subsets

### DISCUSSION THREAD

**Discussion by lrjoshi** [ImmuneDynamics/Spectre#128] [Help]
This package is terrific. I just have a question about scaling expression markers. Is there any function in Spectre or any other method that we can use to change the scale while visualizing the markers ?

In the plot below BV510 is the marker for CD25, which is very rare in PBMC. But as you can see looks like it is expressed in all cell types because the scale ranges from 0.1 to 0.15. What can I do to change the scale, so the expression looks more realistic? Usually, there are 5% CD25 positive cells expressed mainly by CD4 cells. ( Here in the graph PE-Cy7 is CD4, PETr is CD45) ? 

![Multi plot - plotted on UMAP_X by UMAP_Y](https://user-images.githubusercontent.com/35823873/169556263-2553dee0-1f16-4920-9b86-d0b5521c4469.png)

**Comment by AliOsman332** [comment]
Have you gotten an answer? I am also trying to rescale the scale for my plots and was wondering how to do so.

**Comment by ghar1821** [comment]
It is possible to rescale the colours using `make.colour.plot`, but it is a bit hacky. So something like the following:

```
library(Spectre)
library(data.table)

cell.dat = Spectre::demo.clustered

# Just to show what the original Ly6G and CD45 like.
make.colour.plot(cell.dat, "UMAP_X", "UMAP_Y", "Ly6G_asinh")
make.colour.plot(cell.dat, "UMAP_X", "UMAP_Y", "CD45_asinh")

# set a new colour scale for Ly6G based on CD45 colour
# create a new data.table containing the value of CD45, but assign to a column call ly6g
align_col = data.table(Ly6G_asinh=cell.dat$CD45_asinh)

# make new colour plot for Ly6G, but use the colouring scheme for CD45
make.colour.plot(cell.dat, "UMAP_X", "UMAP_Y", "Ly6G_asinh", align.col.by = align_col)
```
The code above will produce the following plots. 

![image](https://github.com/ImmuneDynamics/Spectre/assets/5366317/2475883e-a8b8-4d32-9192-a5916f410bf9)

![image](https://github.com/ImmuneDynamics/Spectre/assets/5366317/fb7e593a-e5f7-4db4-bd68-1b3c3db8c168)

![image](https://github.com/ImmuneDynamics/Spectre/assets/5366317/70fc333a-607d-4258-9487-afc5f1cbff34)

Using the parameter align.col.by, you force colour plot to colour the plot based on the range of values in a new data table `align_col`. The trick here is to have the column Ly6G in `align_col` but give it the values of other column (in this case CD45). 

Hope this helps.

**Reply by AliOsman332** [reply]
I tried introducing your lines. I got negative scalings too for some reason. I am probably doing something wrong. Do you have any tips and or suggestions that can help me more? Or is there a website with detailed explanations? 
Thank you.

**Reply by ghar1821** [reply]
not sure what u meant by negative scaling. can u give an example?
